tests/Silverback.Tests.Performance/MessageValidationLoggingBenchmark.cs

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) 2020 Sergio Aquilini
+// This code is licensed under MIT license (see LICENSE file for details)
+
+using System.Diagnostics.CodeAnalysis;
+using BenchmarkDotNet.Attributes;
+using Silverback.Messaging.Validation;
+using Silverback.Tests.Performance.TestTypes;
+using Silverback.Tests.Types.Domain;
+
+namespace Silverback.Tests.Performance
+{
+    // 3.X.X - MessageValidation Feature
+    //
+    // |                                                                        Method |      Mean |     Error |    StdDev |  Gen 0 |  Gen 1 | Gen 2 | Allocated |
+    // |------------------------------------------------------------------------------ |----------:|----------:|----------:|-------:|-------:|------:|----------:|
+    // |                  CheckMessageIsValid_ValidationModeNone_SinglePropertyInvalid |  6.071 us | 0.1191 us | 0.1854 us | 0.7477 |      - |     - |      5 KB |
+    // |            CheckMessageIsValid_ValidationModeLogWarning_SinglePropertyInvalid |  6.134 us | 0.1213 us | 0.2395 us | 0.7477 |      - |     - |      5 KB |
+    // |                   CheckMessageIsValid_ValidationModeNone_AllPropertiesInvalid |  7.871 us | 0.1560 us | 0.2930 us | 0.9003 |      - |     - |      6 KB |
+    // |             CheckMessageIsValid_ValidationModeLogWarning_AllPropertiesInvalid |  7.992 us | 0.1568 us | 0.2298 us | 0.9003 |      - |     - |      6 KB |
+    // |       CheckMessageIsValid_ValidationModeNone_AllPropertiesInvalidLargeMessage | 15.502 us | 0.3094 us | 0.6458 us | 1.8005 | 0.0153 |     - |     11 KB |
```

Comment:
I didn't expect such a big impact. Could you perform a test with the same "big" message but with only a single validation error (only a single field holding a wrong value) and with a valid one?

(Relatively big, of course.)